DOCS-7696 (OKR3) - Autodiscovery edits, pt 1 (#23416)
DataDog/documentation: content/en/containers/guide/autodiscovery-examples.md

- [Redis integration for all Redis containers](#redis-integration-for-all-redis-containers)
- [Apache integration with HTTP Check for all Apache containers](#apache-integration-and-http-check-on-apache-containers)

For more information about containers and integrations, see [Docker and Integrations][2] and [Kubernetes and Integrations][3].

All examples make use of Datadog's Autodiscovery feature, which allows you to define configuration templates for Agent Checks on designated sets of containers. For more information about Autodiscovery, see [Getting Started with Containers: Autodiscovery][1].

## Redis integration for all Redis containers

In this example scenario, you have a containerized environment in which you want to set up and configure the [Datadog-Redis integration][5] for all containers that match the name `redis`. You may have one container named `redis` and another named `my-custom-redis`, and you want to configure the Redis integration for **both** containers.

The Redis integration comes with [default auto-configuration][4], but you want to additionally specify a `password` parameter and configure log collection. 

> If you were to configure this integration **on a host**, you could reference [`redisdb.d/conf.yaml.example`][6] for parameters and create a `conf.yaml` file that contains the following:

Author: This is not a past tense, but a conditional tense. This page is about configuring on a container (not a host) BUT I want the reader to first imagine hypothetically what it would be like to configure it on a host and then transfer that over to the containerized use case. I'm going to keep the tenses and append the word "hypothetically" to the beginning of the sentence just to make my rhetorical technique more clear.

```
As a result, the Agent contains a `redisdb.yaml` file with the above configuration in the `conf.d` directory.

**Note**: The Redis integration comes with [default auto-configuration][1], which takes precedence over configuration set in Helm values. Because of this, the provided example uses `datadog.ignoreAutoConfig` to disable auto-configuration.

[1]: /containers/guide/auto_conf
{{% /tab %}}
{{< /tabs >}}

All of these examples use [Autodiscovery template variables][7]:
- `%%host%%` is dynamically populated with the container's IP.
- `%%env_REDIS_PASSWORD%%` references an environment variable named `REDIS_PASSWORD` as seen by the Agent process.

## Apache integration and HTTP Check on Apache containers

In this example scenario, you have a containerized environment in which you want to set up and configure the [Datadog-Apache integration][8] for all containers that match the name `apache`. You also want to set up an [HTTP Check][9] for testing two endpoints, `/website_1` and `/website_2`

The Apache integration comes with [default auto-configuration][4], but you want to add an additional configuration: you want to set the [collection interval][11] to 30 seconds.

> If you were to configure this integration **on a host**, you could reference the configuration options in [`apache.d/conf.yaml.example`][10] and [`http_check.d/conf.yaml.example`][12]. You would create two `conf.yaml` files:

Reviewer: ```suggestion
If you configure this integration **on a host**, you can reference the configuration options in [`apache.d/conf.yaml.example`][10] and [`http_check.d/conf.yaml.example`][12]. You can create two `conf.yaml` files:
```
Same as above re: verb tense.

```

{{% /tab %}}
{{% tab "Local file" %}}

* Create the folders `conf.d/` and `conf.d/apache.d` on your host.
* Add the custom auto-configuration below to `conf.d/apache.d/conf.yaml` on your host.

```yaml
ad_identifiers:
  - apache

init_config:

instances:
  - apache_status_url: http://%%host%%/server-status?auto
    min_collection_interval: 30
```

* Next, create the folder `conf.d/http_check.d` on your host.
* Add the custom auto-configuration below to `conf.d/http_check.d/conf.yaml` on your host.

```yaml
ad_identifiers:
  - apache

init_config:

instances:
  - name: "<WEBSITE_1>"
    url: "http://%%host%%/website_1"
    timeout: 1

  - name: "<WEBSITE_2>"
    url: "http://%%host%%/website_2"
    timeout: 1
```

* Finally, mount the host `conf.d/` folder to the containerized Agent `conf.d/` folder.

[1]: https://github.com/DataDog/integrations-core/blob/master/apache/datadog_checks/apache/data/conf.yaml.example

> [2]: /agent/guide/ad_identifiers/

Reviewer: I don't see where you reference these links?

        - apache
      init_config:
      instances:
        - name: "my_website_1"
          url: "http://%%host%%/website_1"
          timeout: 1
        - name: "my_website_2"
          url: "http://%%host%%/website_2"
          timeout: 1
```

**Note**: The Apache integration comes with [default auto-configuration][1], which takes precedence over configuration set in Helm values. Because of this, the provided example uses `datadog.ignoreAutoConfig` to disable auto-configuration.

[1]: /containers/guide/auto_conf

{{% /tab %}}
{{< /tabs >}}

All of these examples use [Autodiscovery template variables][7]:
- `%%host%%` is dynamically populated with the container's IP.

[1]: /getting_started/containers/autodiscovery
[2]: /containers/docker/integrations
[3]: /containers/kubernetes/integrations
[4]: /containers/guide/auto_conf
[5]: /integrations/redis
[6]: https://github.com/DataDog/integrations-core/blob/master/redisdb/datadog_checks/redisdb/data/conf.yaml.example
[7]: /containers/guide/template_variables/
[8]: /integrations/apache
[9]: /integrations/http_check/
[10]: https://github.com/DataDog/integrations-core/blob/master/apache/datadog_checks/apache/data/conf.yaml.example

> [11]: /developers/write_agent_check/#collection-interval

Reviewer: ```suggestion
[11]: /developers/write_agent_check/#updating-the-collection-interval
```
